### PR TITLE
Shared Port fix

### PR DESCRIPTION
**Pull request by AlphaLima**
skip new devices on shared ports
better comparison also for TCP/UDP Client/Server and Bluetooth Connections

### CODE REVIEW

File: Common/Source/Comm/device.cpp [hunk 1]
```diff
@@ -598,45 +627,37 @@ BOOL devInit() {
...
-        // remember: Port1 is the port used by device A, port1 may be Com3 or Com1 etc
+
+	unsigned int j;
+	DeviceList[i].iSharedPort =-1;
+	for( j = 0; j < i ; j++)
+	{
+	  if(!DeviceList[i].Disabled)
+	  if(!DeviceList[j].Disabled)
+	  if( (IsIdenticalPort(i,j)) &&  DeviceList[j].iSharedPort <0)
+	  {
```
Comment (brunotl) [inline Common/Source/Comm/device.cpp]: DeviceList[i].Disabled i always false
no need to call IsIdenticalPort() if DeviceList[j].iSharedPort >= 0

so it's better to write :
```
if( !DeviceList[j].Disabled && (DeviceList[j].iSharedPort < 0) && IsIdenticalPort(i,j) ) {
..
}

```

File: Common/Source/Comm/device.cpp [hunk 2]
```diff
@@ -598,45 +627,37 @@ BOOL devInit() {
...
-          }
-        }
-        UsedPort.insert(Port);
-        
-        // remember: Port1 is the port used by device A, port1 may be Com3 or Com1 etc
+
+	unsigned int j;
+	DeviceList[i].iSharedPort =-1;
+	for( j = 0; j < i ; j++)
+	{
```
Comment (brunotl) [inline Common/Source/Comm/device.cpp]: "j" is local loop iterator so no need to declare them outside

it's better to write 

`for(unsigned j = 0; j < i ; j++)`

File: Common/Source/Comm/device.cpp [hunk 3]
```diff
@@ -550,6 +550,36 @@ static BOOL devInit(PDeviceDescriptor_t d){
...
+  {
+    if (dwIpPort[i] == dwIpPort[j])
+      return true;
+  }
+  else
+  if (_tcscmp(szPort[i], _T("TCPClient")) == 0)
+  {
+    if (_tcscmp(szIpAddress[i], szIpAddress[j]) == 0)
+      if (dwIpPort[i] == dwIpPort[j])
+        return true;
```
Comment (brunotl) [inline Common/Source/Comm/device.cpp]: it's better to test "dwIpPort" before "szIpAddress"

Comment (AlphaLima) [reply Common/Source/Comm/device.cpp]: done:
https://github.com/LK8000/LK8000/pull/1328

File: Common/Source/Comm/device.cpp (hunk 2 above)
Comment (AlphaLima) [reply Common/Source/Comm/device.cpp]: done:
https://github.com/LK8000/LK8000/pull/1328

File: Common/Source/Comm/device.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/Comm/device.cpp]: done:
https://github.com/LK8000/LK8000/pull/1328

**Pull request closed without merging**